### DISCUSSION TITLE
Can some help me on this question ?

### DISCUSSION THREAD

**Discussion by WestFarmer** [etcd-io/etcd#16887] [Q&A] [answered]
https://stackoverflow.com/questions/77437016/why-three-node-etcd-cluster-can-have-low-write-latency

**Comment by jmhbnz** [comment, marked as answer]
Hey @WestFarmer - Thanks for your question. I don't have a stackoverflow account so posting my response here.  

In your question you state:

> I am learning raft. by far, I understand that when client make change state request to to a raft system, leader node need to replicate changes to majority nodes, and at a fixed heartbeat interval.
>
> I assume any client will see a latency bigger than the interval.
>
> While Etcd is a raft implementation, and it's default heartbeat interval was 100ms, I expect a latency greater than 100ms when write a key-value pair to it.

Based on my understanding this is not correct.  The heartbeat interval is the frequency with which the leader will notify followers that it is still the leader.  Not the frequency at which new kv entries are replicated.

You can read more on this here: https://etcd.io/docs/v3.5/tuning/#time-parameters
